elasticsearch_dsl/utils.py

```diff
@@ -111,10 +111,10 @@ def __ne__(self, other: Any) -> bool:
     def __getitem__(self, k: Union[int, slice]) -> Any:
         l = self._l_[k]
         if isinstance(k, slice):
-            return AttrList(l, obj_wrapper=self._obj_wrapper)
+            return AttrList[_ValT](l, obj_wrapper=self._obj_wrapper)  # type: ignore[arg-type]
```

Comment:
This is extremely annoying from mypy. Here we know that `l` is of type `List[_ValT]`, but mypy thinks it can also be a single element of type `_ValT` because it does not see the check for the slice.